android/expoview/src/main/java/host/exp/exponent/kernel/Kernel.kt

```diff
@@ -0,0 +1,1140 @@
+import com.facebook.react.bridge.JavaScriptContextHolder
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReadableMap
+import com.facebook.react.common.LifecycleState
+import com.facebook.react.modules.network.ReactCookieJarContainer
+import com.facebook.react.shell.MainReactPackage
+import com.facebook.soloader.SoLoader
+import de.greenrobot.event.EventBus
+import expo.modules.notifications.service.NotificationsService.Companion.getNotificationResponseFromIntent
+import expo.modules.notifications.service.delegates.ExpoHandlingDelegate
+import expo.modules.updates.manifest.raw.RawManifest
+import host.exp.exponent.*
+import host.exp.exponent.ExpoUpdatesAppLoader.AppLoaderCallback
+import host.exp.exponent.ExpoUpdatesAppLoader.AppLoaderStatus
+import host.exp.exponent.analytics.EXL
+import host.exp.exponent.di.NativeModuleDepsProvider
+import host.exp.exponent.exceptions.ExceptionUtils
+import host.exp.exponent.experience.BaseExperienceActivity
+import host.exp.exponent.experience.ErrorActivity
+import host.exp.exponent.experience.ExperienceActivity
+import host.exp.exponent.experience.HomeActivity
+import host.exp.exponent.headless.InternalHeadlessAppLoader
+import host.exp.exponent.kernel.ExponentErrorMessage.Companion.developerErrorMessage
+import host.exp.exponent.kernel.ExponentKernelModuleProvider.KernelEventCallback
+import host.exp.exponent.kernel.ExponentKernelModuleProvider.queueEvent
+import host.exp.exponent.kernel.ExponentUrls.toHttp
+import host.exp.exponent.kernel.KernelConstants.ExperienceOptions
+import host.exp.exponent.kernel.KernelProvider.instance
+import host.exp.exponent.network.ExponentNetwork
+import host.exp.exponent.notifications.ExponentNotification
+import host.exp.exponent.notifications.ExponentNotificationManager
+import host.exp.exponent.notifications.NotificationActionCenter
+import host.exp.exponent.notifications.ScopedNotificationsUtils
+import host.exp.exponent.storage.ExponentDB
+import host.exp.exponent.storage.ExponentSharedPreferences
+import host.exp.exponent.utils.AsyncCondition
+import host.exp.exponent.utils.AsyncCondition.AsyncConditionListener
+import host.exp.expoview.BuildConfig
+import host.exp.expoview.ExpoViewBuildConfig
+import host.exp.expoview.Exponent
+import host.exp.expoview.Exponent.BundleListener
+import okhttp3.OkHttpClient
+import org.json.JSONException
+import org.json.JSONObject
+import versioned.host.exp.exponent.ExpoTurboPackage
+import versioned.host.exp.exponent.ExponentPackage
+import versioned.host.exp.exponent.ReactUnthemedRootView
+import versioned.host.exp.exponent.modules.api.reanimated.ReanimatedJSIModulePackage
+import java.lang.ref.WeakReference
+import java.util.*
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+// TOOD: need to figure out when we should reload the kernel js. Do we do it every time you visit
+// the home screen? only when the app gets kicked out of memory?
+class Kernel : KernelInterface() {
```

Comment:
For this PR and especially this file, do we want to follow the Kotlin convention of organizing class members? https://kotlinlang.org/docs/coding-conventions.html#class-layout